libs/blocks/email-collection/utils.js

```diff
@@ -0,0 +1,287 @@
+import { createTag, getConfig, getFederatedUrl, localizeLink, loadIms } from '../../utils/utils.js';
+
+const API_ENDPOINTS = {
+  stage: 'https://www.stage.adobe.com/milo-email-collection-api',
+  prod: 'https://www.adobe.com/milo-email-collection-api',
+};
+const FORM_METADATA = {
+  'mps-sname': 'mpsSname',
+  'subscription-name': 'subscriptionName',
+};
+
+export function localizeFederatedUrl(url) {
+  return localizeLink(getFederatedUrl(url), '', true);
+}
+
+const FEDERAL_ROOT = '/federal/email-collection';
+const CONSENT_URL = localizeFederatedUrl(`${FEDERAL_ROOT}/consents/cs4.plain.html`);
+const PLACEHOLDER_URL = localizeFederatedUrl(`${FEDERAL_ROOT}/form-config.json?sheet=placeholders`);
+
+export const FORM_FIELDS = {
+  email: {
+    tag: 'input',
+    attributes: {
+      type: 'email',
+      required: true,
+      autocomplete: 'email',
+    },
+  },
+  'first-name': {
+    tag: 'input',
+    attributes: {
+      type: 'text',
+      readonly: '',
+    },
+  },
+  'last-name': {
+    tag: 'input',
+    attributes: {
+      type: 'text',
+      readonly: '',
+    },
+  },
+  country: {
+    tag: 'input',
+    url: localizeFederatedUrl(`${FEDERAL_ROOT}/form-config.json?sheet=countries`),
+    attributes: {
+      type: 'text',
+      readonly: '',
+    },
+  },
+  organization: {
+    tag: 'input',
+    attributes: {
+      type: 'text',
+      required: true,
+      autocomplete: 'organization',
+    },
+  },
+  occupation: {
+    tag: 'input',
+    attributes: {
+      type: 'text',
+      required: true,
+      autocomplete: 'organization-title',
+    },
+  },
+  state: {
+    tag: 'select',
+    url: getFederatedUrl(`${FEDERAL_ROOT}/form-config.json?sheet=states&limit=2000`),
+    attributes: { required: true },
+  },
+};
+
+export async function getIMS() {
+  if (window.adobeIMS) return window.adobeIMS;
+
+  await loadIms();
+  return window.adobeIMS;
+}
+
+export async function getIMSAccessToken() {
+  try {
+    const ims = await getIMS();
+    const { token } = ims?.getAccessToken() ?? {};
+    return token;
+  } catch (e) {
+    return null;
+  }
+}
+
+export async function getIMSProfile() {
+  try {
+    const ims = await getIMS();
+    const {
+      email,
+      countryCode: country,
+      first_name: firstName,
+      last_name: lastName,
+      ...profile
+    } = await ims.getProfile();
+
+    return {
+      email,
+      country,
+      'first-name': firstName,
+      'last-name': lastName,
+      ...profile,
+    };
+  } catch (e) {
+    return {};
+  }
+}
+
+export function getApiEndpoint(action = 'submit') {
+  const { env } = getConfig();
+  const endPoint = API_ENDPOINTS[env.name] ?? API_ENDPOINTS.stage;
```

Comment:
IMO we should default to production, since it would be a lot worse sending real users to a VPN gated stage experience, compared to a bunch of QEs/authors to production.